sdk/monitor/monitor-opentelemetry/src/utils/common.ts

```diff
@@ -21,6 +22,41 @@ export function ignoreOutgoingRequestHook(request: http.RequestOptions): boolean
   return false;
 }
 
+export const isWindows = (): boolean => {
+  return  process.platform === "win32";
+}
+
+export const isLinux = (): boolean => {
+  return  process.platform === "linux";
+}
+
+/**
+* Get prefix for OS
+* Windows system: "w"
+* Linux system: "l"
+* non-Windows and non-Linux system: "u" (unknown)
+*/
+export const getOsPrefix = (): string => {
+  return isWindows()? "w" : isLinux()? "l" : "u";
+}
+
+/**
+ * TODO: add vm resource provider
```

Comment:
VM resource detector is tricky as it's async (since it makes a call to the azure vm metadata service). The web snippet is initialized as a part of the `useAzureMonitor()` call. In the interest of not making that function async I haven't implemented that detector here.